corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/api.js

```diff
@@ -47,7 +47,16 @@ FormplayerFrontend.module("Menus", function (Menus, FormplayerFrontend, Backbone
                                 response.exception || FormplayerFrontend.Constants.GENERIC_ERROR,
                                 response.type === 'html'
                             );
-                            FormplayerFrontend.trigger('navigation:back');
+
+                            var currentUrl = FormplayerFrontend.getCurrentRoute();
+                            if (FormplayerFrontend.lastError === currentUrl) {
+                                FormplayerFrontend.lastError = null;
+                                FormplayerFrontend.trigger('navigateHome');
```

Comment:
Question: Is 'home' here the root Formplayer page, or is it the application root menu?

If it's possible to do the latter (even if it is only a best-faith effort that isn't always possible, such as in your example where the app home screen IS the error), that would be a big boost to the UX of recovering from this.

Reply:
I look at doing this a bit, and my solution would be to change this line to:
```suggestion
                                var appId = Util.currentUrlToObject()['appId'];
                                if (appId) {
                                    FormplayerFrontend.trigger('app:select', appId);
                                } else {
                                    FormplayerFrontend.trigger('navigateHome');
                                }
```

However, I haven't been able to figure out a way of testing this since I can only get this loop to happen on the home page. I'm going to leave it as is for now since I think that code could cause you to get into a different loop.

Reply:
Agree with that judgement, and it should be a reasonable thing for us to follow up on later if needed.